### PR TITLE
Adjust JS computations

### LINKED ISSUE #77: Fix backwards compatibility for minOrderSize and tickSize
Suggest to change `minOrderSize`and `tickSize` to return UI amounts to stay backwards compatible with v3, and then create new functions for getting BN quantities e.g. `minOrderSizeBN` and `tickSizeBN`

### PR DESCRIPTION
**Pull request by jhlx**
Fixes #77

### CODE REVIEW

File: js/src/bindings.ts [hunk 1]
```diff
@@ -185,14 +185,16 @@ export const placeOrder = async (
 
   const priceFp32 = computeFp32Price(market, limitPrice);
 
-  const formattedLimitPrice =
```
Comment (jhlx) [inline js/src/bindings.ts:188]: Previously had issue with BN being zero when market.baseDecimals > market.quoteDecimals.

File: js/src/market.ts [hunk 2]
```diff
@@ -129,8 +138,13 @@ export class Market {
     this._orderbookAddress = orderbookAddress;
     this._baseSplTokenMultiplier = new BN(10).pow(new BN(baseDecimals));
     this._quoteSplTokenMultiplier = new BN(10).pow(new BN(quoteDecimals));
-    this._tickSize = orderbookState.tickSize.toNumber();
-    this._minOrderSize = marketState.minBaseOrderSize.toNumber();
+    this._tickSize = roundUiAmount(
```
Comment (jhlx) [inline js/src/market.ts:141]: Changed tickSize and minOrderSize to uiAmounts for backwards compatibility with v3. 

Adding 'BN' options.

File: js/src/utils.ts [hunk 3]
```diff
@@ -43,15 +43,33 @@ export const divideBnToNumber = (numerator: BN, denominator: BN): number => {
...
+      fp32Price.toNumber()
+  );
+};
+
+export const computeUiSize = (market: Market, bnSize: BN) => {
+  return bnSize.toNumber() * 10 ** -market.baseDecimals;
+};
+
+export const roundUiAmount = (uiAmount: number) =>
+  Number(uiAmount.toPrecision(5));
```
Comment (jhlx) [inline js/src/utils.ts:59]: This is needed as the 2 ^ 32 division leaves weird rounding artifacts 5+ decimals out.

File: js/src/utils.ts [hunk 4]
```diff
@@ -43,15 +43,33 @@ export const divideBnToNumber = (numerator: BN, denominator: BN): number => {
...
-  const rem = price.umod(tickSize);
-  return price
-    .sub(rem)
-    .mul(new BN(Math.pow(10, market.quoteDecimals - market.baseDecimals)));
+  const remainder = price.umod(market.tickSizeBN);
+  const roundedPrice = price.sub(remainder);
+
+  const scalingExponent = market.quoteDecimals - market.baseDecimals;
+
+  return scalingExponent >= 0
```
Comment (jhlx) [inline js/src/utils.ts:68]: Needed for cases where market.baseDecimals > market.quoteDecimals, to prevent BN from rounding down to 0.

**Comment by dr497** [comment]
Can you resolve the conflicts please?

**Comment by jhlx** [comment]
Closing as no longer relevant with AOB v2.

**Pull request closed without merging**